ComponentKit/StatefulViews/CKStatefulViewReusePool.h

```diff
@@ -10,16 +10,24 @@
 
 #import <UIKit/UIKit.h>
 
+/**
+ Should return YES if the stateful view can be reused, or NO to block reuse of the stateful view.
+ */
+typedef BOOL (^CKStatefulViewReusePoolPendingMayRelinquishBlock)(void);
+
 @interface CKStatefulViewReusePool : NSObject
 
 + (instancetype)sharedPool;
 
+@property (nonatomic, assign) BOOL pendingReusePoolEnabled;
+
 - (UIView *)dequeueStatefulViewForControllerClass:(Class)controllerClass
                                preferredSuperview:(UIView *)preferredSuperview
                                           context:(id)context;
 
 - (void)enqueueStatefulView:(UIView *)view
          forControllerClass:(Class)controllerClass
-                    context:(id)context;
+                    context:(id)context
+         mayRelinquishBlock:(CKStatefulViewReusePoolPendingMayRelinquishBlock)mayRelinquishBlock;
```

Comment:
Well, the view can say NO, and it won't be relinquished, so that's not representative. I think this name is fine. It's just an internal piece of infra, and is descriptive enough.